modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py

```diff
@@ -271,12 +286,25 @@ def width(self):
         int
             The width of the object.
         """
-        if self._width_cache is None:
-            self._width_cache = self.apply(lambda df: len(df.columns))._data
+        self.try_build_width_cache()
         if isinstance(self._width_cache, Future):
             self._width_cache = DaskWrapper.materialize(self._width_cache)
         return self._width_cache
 
+    def try_build_width_cache(self) -> Union[Future, int]:
```

Comment:
Shouldn't this and the length one just be `build_..._cache`?

Reply:
changed